ts2mls/js/src/main/scala/ts2mls/JSWriter.scala

```diff
@@ -7,7 +7,8 @@ import mlscript.utils._
 
 class JSWriter(filename: String) {
   import JSWriter._
-  private val out = fs.openSync(filename, "rs+")
+  private val out =
+    if (fs.existsSync(filename)) fs.openSync(filename, "rs+") else fs.openSync(filename, "w+")
```

Comment:
What does this do? Can you add some comments to explain?

Reply:
the mode `rs+` would fail when the file does not exist, so it can not be used to generate new files or tests

the mode `w+` would lead to modification of test files even though nothing has benn changed, thus the diffTests would failed too

Reply:
I just want to know what these things mean! I had to google this myself, which was pretty painful, to find this page: https://nodejs.org/api/fs.html#file-system-flags. You could have included the link.

Why do you need synchronous uncached writing? The doc emphasizes it has bad performance. _These_ are what I'm looking for in code documentation. Think about the people who'll read this code after you wrote it!

Reply:
You are right! I removed `s` in `rs+` and added the link in the comment.